### ISSUE TITLE
Internal error occured: DecodeError "(in 'passphrase' field) suffix is not in base-16 format: still spatial equal spell ticket method cause shine future flavor inflict huge"

### ISSUE THREAD

**Issue by tanzeelrana** [input-output-hk/cardano-sl#2036]
My curl request :

`curl -X POST "https://localhost:8090/api/wallets/new?passphrase=still%20spatial%20equal%20spell%20ticket%20method%20cause%20shine%20future%20flavor%20inflict%20huge" -H "accept: application/json;charset=utf-8" -H "Content-Type: application/json;charset=utf-8" -d "{ \"cwInitMeta\": { \"cwName\": \"string\", \"cwAssurance\": \"CWAStrict\", \"cwUnit\": 0 }, \"cwBackupPhrase\": { \"bpToList\": [ \"string\" ] }}"`

Response : 

Internal error occured: DecodeError "(in 'passphrase' field) suffix is not in base-16 format: still spatial equal spell ticket method cause shine future flavor inflict huge"

**Comment by adinapoli-iohk** [comment]
Hey @tanzeelrana !

The `passphrase` is something different from what you may think immediately (@akegalj , @Martoon-00 , this is yet another testament that our current API is confusing). A `passphrase` is simply an (optional) extra password you specify when creating a wallet which will protect your private keys when spending funds. It is _not_ the backup phrase, which is instead a 12-words mnemonic as the one you tried to post (look that you will still need to pass a 12-word mnemonic as part of the `bpToList` field, that in your example is simply set to the `["string"]` array.

In particular, if you want to specify a passphrase (think of it as a _spending password_ needed when generating a new address or spending funds) there are rules you need to abide to: it must be exactly 32-chars long and be encoded in base-16. You can use a tool like [this one](http://www.simplycalc.com/base16-encode.php) to play around.

**Comment by tanzeelrana** [comment]
@adinapoli-iohk  

lets see if i clearly understood this. so the 12 word phrase needs to be converted into a single 32-chars long and be encoded in base-16 string and then be passed ?

**Comment by tanzeelrana** [comment]
Aha !! sweet got it !! :) 

I have one more question. 
How do I generate a 12 word phrase that has not been used previously ???

**Comment by akegalj** [comment]
@tanzeelrana
You can use some od generic tools for generatong bip39 12 word phrases like https://iancoleman.io/bip39/ (most of popular languages have this implemented)

Chances of collision are very minimal https://www.google.hr/amp/s/amp.reddit.com/r/ledgerwallet/comments/6cjvam/probability_of_seed_phrase_collisionbrute_forcing/

**Comment by alvinoktavbitcoin** [comment]
hi I don't really understand about the words that I need to enter, can you please tell me the steps more clearly ? I really hope

**Comment by adinapoli-iohk** [comment]
@alvinoktavbitcoin As @akegalj said, you can simply go to a website like https://iancoleman.io/bip39/ and use their web interface to generate a 12-words mnemonic password which you can use to initialise an HD wallet. Bear in mind if you lose this password you won't be able to restore your wallet anymore, so be careful 😉

**Comment by alvinoktavbitcoin** [comment]
yes , I have generate 12 words, now what I should I do with that 12 words ?
I want to use API : https://cardanodocs.com/technical/wallet/api/#/ , but
its need me to enter a passphrase that base 16-code

On Fri, Jan 12, 2018 at 4:41 PM, Alfredo Di Napoli <notifications@github.com

**Comment by cleverca22** [comment]
there is no client certificates, the `ca.crt` file is to tell the client what root to trust, so the self-signed cert is accepted

under windows, its in a path somewhere close to `c:\program files\daedalus\tls\ca\ca.crt`

**Comment by afzydev** [comment]
@tanzeelrana 

Please tell me how to run generate new address api.
Thanks

**Comment by sunujacob** [comment]
I generate 12 random words using [bip39](https://iancoleman.io/bip39/#english) ? What should I do next? are there tools to generate passphrase that `  must be exactly 32-chars long and be encoded in base-16.` ?

**Comment by sunujacob** [comment]
Is below api call correct?
`
    curl -k --request POST \
       --url https://127.0.0.1:8090/api/wallets/new \
       --header 'content-type: application/json' \
       --data '{ 
    "cwInitMeta": {
      "cwName": "My lovely wallet",
      "cwAssurance": "CWANormal",
      "cwUnit": 0
    },
    "cwBackupPhrase": {
        "bpToList":
        ["squirrel","material","silly","twice","direct","slush","pistol","razor","become","junk","kingdom","flee"]
      }
    }'
`

**Comment by Martoon-00** [comment]
Yes, it seems correct.
Mnemonics should be generated according to bip39, right.
Passphrase is optional; to make it complaint with daedalus you need to first evaluate its `blake2b` hash, getting 32-length string, and then encode to `base16` before making request. See https://goo.gl/2uopZx

**Comment by lhfly5201314** [comment]
@tanzeelrana Hi,friend!Could you paste your curl command about create wallet with passphrase and create addresses?I am really confused about the api docs!
And which version api are u using?I tried the v1,but it didn't work,how about u?
